docs/types.rst

```diff
@@ -753,14 +756,17 @@ non-persistent area where function arguments are stored, and behaves mostly like
     depending on the kind of variable, function type, etc., but all complex types must now give an explicit
     data location.
 
+.. _data-location-assignment:
+
+Data location and assignment behaviour
```

Comment:
Please split this and the above part into a new PR.

Reply:
Will do, but adding the RST links is needed for the sections below, so I'll leave those in.